enginetest/queries/script_queries.go

```diff
@@ -2252,6 +2252,31 @@ var ScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "basic test on tables dual and `dual`",
+		SetUpScript: []string{
+			"CREATE TABLE `dual` (id int)",
+			"INSERT INTO `dual` VALUES (2)",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "SELECT * from `dual`;",
+				Expected: []sql.Row{{2}},
+			},
+			{
+				Query:    "SELECT 3 from dual;",
+				Expected: []sql.Row{{3}},
+			},
+			{
+				Query:       "SELECT * from dual;",
+				ExpectedErr: sql.ErrNoTablesUsed,
+			},
+			{
+				Query:    "select 0 as dummy, (select dummy);",
+				Expected: []sql.Row{{0, 0}},
+			},
```

Comment:
Awesome! Very cool that this query will work now!! 🎉 

While you're in here could you unskip this query in the `column_alias_queries.go` file, too? 
https://github.com/dolthub/go-mysql-server/blob/main/enginetest/queries/column_alias_queries.go#L228

There's also a second query in that same file, right underneath the one above, that is getting an incorrect "x" value from the dual table. Could you check and see if it happens to work with your latest changes? I suspect there's another problem with properly resolving the union expression and not processing the opaque node correctly somewhere, but would be awesome if I was wrong that query just started working, too. 😄